### ISSUE TITLE
Feature: Safe way of using IDs in create/update statements

### ISSUE THREAD

**Issue by williamoverton** [surrealdb/surrealdb.js#164]
### Is your feature request related to a problem?

If I wanted to update a record by its ID, but the id is only known at runtime, I am forced to do string concatenation 

```javascript
await client.update(`user:${req.body.username}`, {
  username: username,
  password: password
});
```

This can lead to the user editing the statement to be operating on a completely different type of record, for example if the username input was set to "something:else" I would be adding a colon which is treated differently.

I think the currently implementation could lead to a lot of developers inadvertently leaving SQL injection vulnerabilities in their code. 

### Describe the solution

Option 1: Allow for specifying IDs in a way that would not be treated specially
Option 2: Allow for throwing an error if an invalid ID is passed in.

### Alternative methods

Allow for prepared statement style variable passing:

```
await client.update(`user:$username`, {
    first_name: firstname,
    last_name: lastname
}, {
    username: username
})
```

### SurrealDB version

1.0.0-beta.11+20230905.59718fc for windows on x86_64

### SurrealDB.js version

0.8.4

### Contact Details

willovertonuk@gmail.com

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kearfy** [comment]
Hi @williamoverton, while this would absolutely be a cool feature, it's currently not possible to implement this unfortunately! This is a limitation within the Websocket text protocol.

It is possible to execute a custom query and to pass along an ID like you want to however!

```js
const [updated_record] = await client.query(
    /* surql */ `
        UPDATE ONLY type::thing('user', $username) 
            SET first_name = $firstname, last_name = $lastname;
    `, 
    {
        first_name: firstname,
        last_name: lastname,
        username,
    }
)
```

PS, one suggestion for your schema: I'd personally advice to not set the username as the record ID, as record IDs are immutable 😃

**Comment by kearfy** [comment]
This should be resolved by #227, with the introduction of the new RecordId class
